### DISCUSSION TITLE
How to limit the camera zoom?

### DISCUSSION THREAD

**Discussion by bjlittle** [pyvista/pyvista#2587] [Q&A] [answered]
Is it possible to limit the depth of interactive zooming for the camera in a scene?

e.g., Given a simple scene containing a sphere mesh of radius=0.5 centred around the origin 
```python
import pyvista as pv

plotter = pv.Plotter()
plotter.add_mesh(pv.Sphere(), show_edges=True)
plotter.show_grid()
plotter.show()
```
![image](https://user-images.githubusercontent.com/2051656/167893671-5608a937-0641-487a-b505-5606435a89a8.png)

How do you **prevent** the user interactively zooming the camera beyond the surface and into the sphere?

i.e., limit the camera zooming to a radius of 0.5 around the origin so that the surface is always visible... is this possible?

**Comment by banesullivan** [comment, marked as answer]
This could get complicated.

I'd create an event handler on the `Camera` that checks the focal point and position of the camera. If the focal point is inside the sphere, then make sure the position stays just outside the sphere.

Or perhaps you could implement a custom interaction style that handles this sort of logic.
